launchable/commands/split_subset.py

```diff
@@ -180,18 +250,71 @@ def run(self):
                         err=True)
                     return
 
-            if len(output) == 0:
-                click.echo(click.style("Error: no tests found in this subset id.", 'yellow'), err=True)
-                return
+        def _write_split_by_groups_group_names(self, subset_group_names: List[str], rest_group_names: List[str]):
+            if is_output_exclusion_rules:
+                subset_group_names, rest_group_names = rest_group_names, subset_group_names
+
+            if len(subset_group_names) > 0:
+                with open("{}/{}".format(split_by_groups_output_dir, SPLIT_BY_GROUP_SUBSET_GROUPS_FILE_NAME),
+                          "w+", encoding="utf-8") as f:
+                    f.write("\n".join(subset_group_names))
+
+            if is_split_by_groups_with_rest:
+                with open("{}/{}".format(split_by_groups_output_dir, SPLIT_BY_GROUP_REST_GROUPS_FILE_NAME),
+                          "w+", encoding="utf-8") as f:
+                    f.write("\n".join(rest_group_names))
+
+        def split_by_group_namess(self):
+            try:
+                res = client.request("POST", "{}/split-by-groups".format(subset_id))
+                res.raise_for_status()
+
+                is_observation = res.json().get("isObservation", False)
+                split_groups = res.json().get("splitGroups", [])
+
+                subset_group_names = []
+                rest_group_names = []
+
+                for group in split_groups:
+                    group_name = group.get("groupName", "")
+                    subset = group.get("subset", [])
+                    rests = group.get("rest", [])
+
+                    if is_observation:
+                        subset, rests = subset + rests, []
 
-            if is_observation:
-                output = output + rests
-            if rest:
-                if len(rests) == 0:
-                    rests.append(output[0])
+                    if len(subset) > 0 and group_name != SPLIT_BY_GROUPS_NO_GROUP_NAME:
+                        subset_group_names.append(group_name)
 
-                self.write_file(rest, rests)
+                    if group_name != SPLIT_BY_GROUPS_NO_GROUP_NAME:
```

Comment:
`elif`?

Reply:
When the `--output-exclusion-rules` option enable, subset group and rest group will switch. Then, there is rest data, but if it's not in rest_group_names, it's difficult to create subset_group_names.txt
So I added filter logic https://github.com/launchableinc/cli/pull/502/files/8bd7e154a0063632b2a3cf74640024c442cf89c9#diff-36cadddae14cddbb1f42e166c7cb78bb91deeaf3761052e9438363d14dd2c2cbR257-R260

Reply:
I misunderstood so I fixed